components/dot_menu/index.ts

```diff
@@ -62,7 +64,8 @@ function mapStateToProps(state: GlobalState, ownProps: Props) {
     const currentTeamUrl = `${getSiteURL()}/${currentTeam.name}`;
 
     const apps = appsEnabled(state);
-    const appBindings = apps ? getAppBindings(state, AppBindingLocations.POST_MENU_ITEM) : [];
+    const showBindings = apps && !isSystemMessage(post) && !isCombinedUserActivityPost(post.id);
+    const appBindings = showBindings ? getAppBindings(state, AppBindingLocations.POST_MENU_ITEM) : [];
```

Comment:
This is going to cause performance issues when `showBindings` is false since it's going to pass a new empty array every time `mapStateToProps` is called (which happens constantly). We should either pass an undefined value and have `DotMenu` handle that case itself (potentially by using `defaultProps` to set `appBindings` to an empty array) or we should wrap this in a per-component memoized selector. I'd recommend the `defaultProps` approach here since it's probably much easier.

Reply:
@hmhealey Would you happen to know how to make the type check error go away, when passing in `undefined` for a required prop, when the prop is indeed being set in `defaultProps`?

Here's a PR to solve performance issues via memoized selectors https://github.com/mattermost/mattermost-webapp/pull/7775